peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+introduces concern with this behavior. PEP 708 added a way to restrict
+installers from drawing packages from unexpected, potentially insecure
+indexes. On the other hand, the term “best-match” is technically
+correct, but also misleading. The “best match” varies by user and by
+application. “Best” is technically correct in the sense that it is a
+global optimum according to the match criteria specified above, but that
+is not necessarily what is “best” in a user’s eyes. “Version priority”
+is a proposed term that avoids the concerns with the uv terminology,
+while approximating the behavior in the most user-identifiable way that
+packages are compared.
+
+“Index priority”
+----------------
+
+Relative to “version priority” behavior, this strategy does not collate
+the contents of each index before evaluating the best match. Where the
+best-match behavior seeks a global best match across all repositories,
+the first-match strategy seeks a local best match within one repo, and
+accepts the first match that it finds. The uv and PDM projects have led
+the way in experimenting with different ways of dealing with multiple
+package sources. Uv’s default behavior is one that implements
+first-match behavior (and coins the terms “first-match” and
+“best-match”). In general, the strategy evaluates best match on the
+top-priority repo, and proceeds to subsequent repos only if the current
+package request has no viable candidates. The criteria and process for
+evaluating “best match” is the same for both index priority and version
+priority. The difference is that version priority combines all indexes
+before ordering and prioritizing them, while index priority iterates
+over the indexes in user-specified order, stopping at the first matching
+distribution. The index (or “source” in PDM terms) priority is
+explicitly defined by a configuration hierarchy. Each package’s finder
+starts at the beginning of the list of repositories, so each package
+starts over with the repository list. In other words, if one package has
+no valid candidates on the first index, but finds a hit on the second
+index, subsequent packages still start their search on the first index,
+rather than starting on the second.
+
+One desirable behavior that the index priority strategy implies is that
+there are no “surprise” updates, where a version bump on a
+lower-priority repo wins out over a curated, approved higher-priority
+repo. This is related to the security improvement of PEP 708, where
+packages can restrict the external repos that distributions can come
+from, but it is more configurable by end users. The package installs are
+only expected to change when either the higher-priority repo or the
+index priority configuration change. This stability and predictability
+makes it viable to configure indexes as a more persistent property of an
+environment, rather than a one-off argument for one install command.
+
+One important implementation detail of index priority is that caching
```

Comment:
Don't lockfiles have the same implications?

Reply:
Definitely. Added.